File: primer.rb
class Integer
  def prime?(known_primes)
    sqrt = Math.sqrt(self)
    
    known_primes.each do |x|
      break if x > sqrt
      return false if self % x == 0
    end
    
    true
  end
  
  def self.primes(max)
    primes = []

    (2..max).each do |x|
      primes << x if x.prime?(primes)
    end

    primes
  end
end

t1 = Time.new
primes = Integer.primes(100000)
t2 = Time.new

puts primes.join(' ')
puts "Found #{primes.count} prime numbers in #{(t2 - t1)} sec."

File: primer.js
var Primer = {
  isPrime: function(number, known_primes) {
    var sqrt = Math.sqrt(number);
    
    for (var i = 0, l = known_primes.length, x; i < l; i++) {
      x = known_primes[i];
      if (x > sqrt) break;
      if (number % x == 0) return false;
    }
    
    return true;
  },
  
  getPrimes: function(max) {
    var primes = [];
    
    for (i = 2; i <= max; i++) {
      if (Primer.isPrime(i, primes)) primes.push(i);
    }
    
    return primes;
  }
};

t1 = new Date().getTime();
primes = Primer.getPrimes(100000);
t2 = new Date().getTime();

print(primes.join(' '));
print("Found " + primes.length + " prime numbers in " + (t2 - t1)/1000 + " sec.");